### ISSUE TITLE
[HOLD #24444] Ensure CI builds are tested when modified in a PR

### ISSUE THREAD

**Issue by Julesssss** [Expensify/App#25493]
@Expensify/mobile-deployers -- does anyone have any better ideas?

### Problem

We depend on the AdHoc test script to speed up testing. [Occasionally](https://github.com/Expensify/App/pull/24306#issuecomment-1684011459) the script is modified without being tested, which can delay testing on many issues, including those which we depend on for unblocking the App deployment. 

### Solution

Checks for modifications of the CI files and add a comment asking people to manually run the build
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01af8b1c07f6d15580</li>
        <li>Upwork Job ID: 1692548383326777344</li>
        <li>Last Price Increase: 2023-08-18</li>
    </ul>
</details>

**Comment by roryabraham** [comment]
@Julesssss maybe we can put this on HOLD for https://github.com/Expensify/App/issues/24444 ?

**Comment by Julesssss** [comment]
Sounds good to me 👍
